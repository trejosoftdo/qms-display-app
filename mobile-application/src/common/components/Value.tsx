import { FC } from 'react';
import { StyleSheet, View } from 'react-native';
import { Avatar, Text } from 'react-native-paper';
import AppLightTheme from '../theme';

/**
 * ValueProps defines the props for the Value Component.
 *
 * @property icon - icon to display
 * @property value - value to display
 */
interface ValueProps {
  icon: string;
  value: string;
}


/**
 * A component that represents a value with icon
 *
 * @param {ValueProps} props - The props for the Value component.
 */
const Value: FC<ValueProps> = (props: ValueProps) => (
  <View style={styles.container}>
    <Avatar.Icon style={styles.icon} icon={props.icon} /> 
    <Text style={styles.number} variant="headlineLarge">
      {props.value}
    </Text>
  </View>
);

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    borderWidth: 1,
    borderColor: AppLightTheme.colors.outline,
    padding: 16,
    borderRadius: 8,
    marginBottom: 32,
    width: '80%',
    maxWidth: 280,
  },
  icon: {
    backgroundColor: AppLightTheme.colors.primary,
  },
  number: {
    color: AppLightTheme.colors.secondary,
    fontWeight: '600',
    paddingTop: 12,
  },
});

export default Value;
