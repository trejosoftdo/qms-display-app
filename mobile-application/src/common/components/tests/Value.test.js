import React from 'react';
import Value from '../Value';


describe('Value component', () => {
  const props = {
    icon: 'lock',
    value: 'Test value',
  };

  it('renders the component with the provided props as expected', () => {
    expect(renderShallow(<Value {...props} />)).toMatchSnapshot();
  });
});