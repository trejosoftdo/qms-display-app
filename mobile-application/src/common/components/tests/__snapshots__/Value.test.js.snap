// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Value component renders the component with the provided props as expected 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderColor": "#d4d4d4",
      "borderRadius": 8,
      "borderWidth": 1,
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
      "marginBottom": 32,
      "maxWidth": 280,
      "padding": 16,
      "width": "80%",
    }
  }
>
  <Avatar.Icon
    icon="lock"
    style={
      {
        "backgroundColor": "#2929cc",
      }
    }
  />
  <ForwardRef(Text)
    style={
      {
        "color": "#3c3744",
        "fontWeight": "600",
        "paddingTop": 12,
      }
    }
    variant="headlineLarge"
  >
    Test value
  </ForwardRef(Text)>
</View>
`;
